R/ag_production_technology.R

```diff
@@ -57,7 +57,10 @@ AgProductionTechnology_initCalc <- function(aLandLeaf, aPeriod, aScenarioInfo) {
       # Adjust last period's yield by tech change
       aLandLeaf$mYield[aPeriod] <- preYield * ((1 + aLandLeaf$mAgProdChange[[aPeriod]]) ^ timestep)
     }
-  } else {
+  } else if (length(aLandLeaf$mYield) > 0) {
```

Comment:
FWIW, the combination of convoluted control paths and dense formatting makes this entire function very hard to understand.  Generally speaking, when you find yourself writing something like this, you should be asking if there is some kind of refactoring you could do.